### ISSUE TITLE
Address user dissatisfaction for memory screen.

### ISSUE THREAD

**Issue by polina-c** [flutter/devtools#6972]
- [ ] Information should not be lost when app disconnected/connected or app crash
- [ ] Import/Export snapshots to enable: analysis of prod snapshots, sharing with team, analysis with other tools
- [ ] Handle memory limit
    - [ ] Design doc

**Comment by polina-c** [comment]
@kenzieschmoll , can you link the doc that gives details about user feedback for memory screen?

**Comment by polina-c** [comment]
motivation: b/316403803 (internal link)

**Comment by mraleph** [comment]
I don't know if this warrants a separate issue or not but I have tried using DevTools to debug https://github.com/flutter/flutter/issues/139650 and snapshotting just hang on me - and the heap was barely 100Mb. So in the end I could not use snapshot tab to solve it. I managed to use instance tracing though to figure it out, but the experience was poor (especially when you click snapshotting button and the whole tab just hangs on you).

**Comment by polina-c** [comment]
> I don't know if this warrants a separate issue or not but I have tried using DevTools to debug [flutter/flutter#139650](https://github.com/flutter/flutter/issues/139650) and snapshotting just hang on me - and the heap was barely 100Mb. So in the end I could not use snapshot tab to solve it. I managed to use instance tracing though to figure it out, but the experience was poor (especially when you click snapshotting button and the whole tab just hangs on you).

Thank you for reporting! You are using mac, right? What are CPU and memory numbers on the mac? How do you know that heap was 100Mb?

**Comment by mraleph** [comment]
> You are using mac, right? 

Yes. It's an M1 Mac Book Pro with 32 Gb of RAM. 

> How do you know that heap was 100Mb?

That's what devtools was showing me when I tried to take snapshot.

**Comment by polina-c** [comment]
> That's what devtools was showing me when I tried to take snapshot.

On the chart, right?

**Comment by mraleph** [comment]
> On the chart, right?

Yep, also on the table where various classes are listed together with instance count and instance sizes.

**Comment by polina-c** [comment]
@mraleph , performance of snapshotting is significantly improved on flutter master
However, Chrome tab has limit 4 GB and will crash out of memory if a snapshot size is 3GB or more.

So, you will need to [build desktop DevTools](https://github.com/flutter/devtools/blob/master/BETA_TESTING.md) to workaround this.

I will be happy to help if you experience difficulties. Tweet me at @PolinaLinaC to setup video call.

**Comment by polina-c** [comment]
done
